sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -144,18 +137,10 @@ def __init__(self, gcs_base, target_image):
   def _check_gcs_path(self, gcs_path):
     """ _check_gcs_path check both the path validity and write permissions """
     logging.info('Checking path: {}...'.format(gcs_path))
-    if gcs_path.startswith('gs://'):
-      with tempfile.TemporaryDirectory() as local_build_dir:
-        rand_filename = str(uuid.uuid4()) + '.txt'
-        local_rand_file = os.path.join(local_build_dir, rand_filename)
-        Path(local_rand_file).touch()
-        random_gcs_bucket = os.path.join(gcs_path, rand_filename)
-        GCSHelper.upload_gcs_file(local_rand_file, random_gcs_bucket)
-        GCSHelper.remove_gcs_blob(random_gcs_bucket)
-        return True
-    else:
+    if not gcs_path.startswith('gs://'):
```

Comment:
No need for the read/write permission pre-check because GCSHelper will raise meaningful exceptions.